kernel/register.cc

```diff
@@ -696,18 +718,18 @@ void Backend::extra_args(std::ostream *&f, std::string &filename, std::vector<st
 				delete ff;
 				log_cmd_error("Can't open output file `%s' for writing: %s\n", filename.c_str(), strerror(errno));
 			}
+			filenames.push_back(filename);
 			f = ff;
 		}
 	}
 
-	if (called_with_fp)
-		args.push_back(filename);
```

Comment:
Why was this removed? This actually changes behavior in potentially breaking ways and seems unrelated to the actual changes done to remove empty files on cmd errors.

Reply:
reverted it